diff --git a/dist/index.mjs b/dist/index.mjs
index 5d684e675461ef2e6a31ca70b59e9fbc0cfa1c71..b70ddea8f56b9e95ea3a9b9bf473764a6c336dee 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -163,7 +163,14 @@ const enforceCatalog = createEslintRule({
             type: "boolean",
             description: "Whether to reuse existing catalog when moving version to catalog with autofix",
             default: true
-          }
+          },
+          skipPackages: {
+            type: "array",
+            description: "A list of packages to skip when enforcing the catalog",
+            items: {
+              type: "string"
+            }
+          },
         },
         additionalProperties: false
       }
@@ -178,9 +185,12 @@ const enforceCatalog = createEslintRule({
       allowedProtocols = ["workspace", "link", "file"],
       defaultCatalog = "default",
       autofix = true,
-      reuseExistingCatalog = true
+      reuseExistingCatalog = true,
+      skipPackages = []
     } = options || {};
     for (const { packageName, specifier, property } of iterateDependencies(context)) {
+      if (skipPackages.includes(packageName))
+        continue;
       if (specifier.startsWith("catalog:"))
         continue;
       if (allowedProtocols?.some((p) => specifier.startsWith(p)))
