diff --git a/dist/index.mjs b/dist/index.mjs
index 5d684e675461ef2e6a31ca70b59e9fbc0cfa1c71..e197476c690d924e3f3e1a40c0f949408fd41d9b 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -163,7 +163,14 @@ const enforceCatalog = createEslintRule({
             type: "boolean",
             description: "Whether to reuse existing catalog when moving version to catalog with autofix",
             default: true
-          }
+          },
+          skip: {
+            type: "array",
+            description: "A list of dependency package names to skip enforcing catalog",
+            items: {
+              type: "string"
+            }
+          },
         },
         additionalProperties: false
       }
@@ -178,9 +185,12 @@ const enforceCatalog = createEslintRule({
       allowedProtocols = ["workspace", "link", "file"],
       defaultCatalog = "default",
       autofix = true,
-      reuseExistingCatalog = true
+      reuseExistingCatalog = true,
+      skip = []
     } = options || {};
     for (const { packageName, specifier, property } of iterateDependencies(context)) {
+      if (skip.includes(packageName))
+        continue;
       if (specifier.startsWith("catalog:"))
         continue;
       if (allowedProtocols?.some((p) => specifier.startsWith(p)))
