diff --git a/lib/ConfigCache.js b/lib/ConfigCache.js
index 55d07a04c611f65e0a27f59b5de9c89c9d393888..e15e58724147c0f6eacfd5b01a4ce995d867ddcd 100644
--- a/lib/ConfigCache.js
+++ b/lib/ConfigCache.js
@@ -29,6 +29,76 @@ exports.ConfigCache = void 0;
 const tsdoc_config_1 = require("@microsoft/tsdoc-config");
 const path = __importStar(require("path"));
 const Debug_1 = require("./Debug");
+
+function _nullishCoalesce(lhs, rhsFn) { if (lhs != null) { return lhs; } else { return rhsFn(); } }
+var _fs = require('fs');
+var _path = require('path');
+var MAX_PATH_SEARCH_DEPTH = 30;
+var depth = 0;
+
+const ROOT_FILES = [
+  "storm-workspace.json",
+  "storm-workspace.json",
+  "storm-workspace.yaml",
+  "storm-workspace.yml",
+  "storm-workspace.js",
+  "storm-workspace.ts",
+  ".storm-workspace.json",
+  ".storm-workspace.yaml",
+  ".storm-workspace.yml",
+  ".storm-workspace.js",
+  ".storm-workspace.ts",
+  "lerna.json",
+  "nx.json",
+  "turbo.json",
+  "npm-workspace.json",
+  "yarn-workspace.json",
+  "pnpm-workspace.json",
+  "npm-workspace.yaml",
+  "yarn-workspace.yaml",
+  "pnpm-workspace.yaml",
+  "npm-workspace.yml",
+  "yarn-workspace.yml",
+  "pnpm-workspace.yml",
+  "npm-lock.json",
+  "yarn-lock.json",
+  "pnpm-lock.json",
+  "npm-lock.yaml",
+  "yarn-lock.yaml",
+  "pnpm-lock.yaml",
+  "npm-lock.yml",
+  "yarn-lock.yml",
+  "pnpm-lock.yml",
+  "bun.lockb"
+];
+const ROOT_DIRECTORIES = [
+  ".storm-workspace",
+  ".nx",
+  ".github",
+  ".vscode",
+  ".verdaccio"
+];
+
+function findWorkspaceRoot(startPath) {
+  const _startPath = _nullishCoalesce(startPath, () => ( process.cwd()));
+  if (ROOT_DIRECTORIES.some(
+    (endDirName) => _fs.existsSync.call(void 0, _path.join.call(void 0, _startPath, endDirName))
+  )) {
+    return _startPath;
+  }
+  if (ROOT_FILES.some(
+    (endFileName) => _fs.existsSync.call(void 0, _path.join.call(void 0, _startPath, endFileName))
+  )) {
+    return _startPath;
+  }
+  if (_startPath !== "/" && depth++ < MAX_PATH_SEARCH_DEPTH) {
+    const parent = _path.join.call(void 0, _startPath, "..");
+    return findWorkspaceRoot(parent);
+  }
+  return void 0;
+}
+
+
 // How often to check for modified input files.  If a file's modification timestamp has changed, then we will
 // evict the cache entry immediately.
 const CACHE_CHECK_INTERVAL_MS = 3 * 1000;
@@ -45,6 +115,7 @@ class ConfigCache {
         const [seconds, nanoseconds] = process.hrtime();
         return seconds * 1000 + nanoseconds / 1000000;
     }
+
     static getForSourceFile(sourceFilePath, tsConfigRootDir) {
         const sourceFileFolder = path.dirname(path.resolve(sourceFilePath));
         // First, determine the file to be loaded. If not found, the configFilePath will be an empty string.
@@ -56,6 +127,7 @@ class ConfigCache {
         // (instead of an empty string)
         const cacheKey = configFilePath || sourceFileFolder + '/';
         Debug_1.Debug.log(`Cache key: "${cacheKey}"`);
+
         const nowMs = ConfigCache._getTimeInMs();
         let cachedConfig = undefined;
         // Do we have a cached object?
@@ -81,12 +153,219 @@ class ConfigCache {
                 }
             }
         }
+
+        // Load the object
+        if (!cachedConfig) {
+            if (ConfigCache._cachedConfigs.size > CACHE_MAX_SIZE) {
+                Debug_1.Debug.log('Clearing cache');
+                ConfigCache._cachedConfigs.clear(); // avoid a memory leak
+            }
+            const configFile = tsdoc_config_1.TSDocConfigFile.loadFile(configFilePath);
+            if (configFile.fileNotFound) {
+                Debug_1.Debug.log(`File not found: "${configFilePath}"`);
+            }
+            else {
+                Debug_1.Debug.log(`Loaded: "${configFilePath}"`);
+            }
+
+            cachedConfig = {
+                configFile,
+                lastCheckTimeMs: nowMs,
+                loadTimeMs: nowMs
+            };
+            ConfigCache._cachedConfigs.set(cacheKey, cachedConfig);
+        }
+
+        return cachedConfig.configFile;
+    }
+
+    static _checkConfigFilePath(workspaceRoot, configFilePath) {
+      const workspaceRootPath = workspaceRoot ? workspaceRoot : "./";
+      if (!configFilePath) {
+        return undefined;
+      }
+
+      if (_fs.existsSync(configFilePath)) {
+        return configFilePath;
+      } else if (
+        _fs.existsSync(
+          path.join(
+            workspaceRootPath,
+            configFilePath
+          )
+        )
+      ) {
+        return path.join(
+          workspaceRootPath,
+          configFilePath
+        );
+      } else if (
+        _fs.existsSync(
+          path.join(
+            workspaceRootPath,
+            "node_modules",
+            configFilePath
+          )
+        )
+      ) {
+        return path.join(
+          workspaceRootPath,
+          "node_modules",
+          configFilePath
+        );
+      } else if (
+        _fs.existsSync(
+          path.join(
+            workspaceRootPath,
+            "node_modules",
+            "@storm-software",
+            "tsdoc",
+            configFilePath
+          )
+        )
+      ) {
+        return path.join(
+          workspaceRootPath,
+          "node_modules",
+          "@storm-software",
+          "tsdoc",
+          configFilePath
+        );
+      } else if (
+        _fs.existsSync(
+          path.join(
+            workspaceRootPath,
+            "node_modules",
+            "@storm-software",
+            "tsdoc",
+            "config",
+            configFilePath
+          )
+        )
+      ) {
+        return path.join(
+          workspaceRootPath,
+          "node_modules",
+          "@storm-software",
+          "tsdoc",
+          "config",
+          configFilePath
+        );
+      } else if (
+        _fs.existsSync(
+          path.join(
+            workspaceRootPath,
+            "node_modules",
+            "@storm-software",
+            "tsdoc",
+            "dist",
+            configFilePath
+          )
+        )
+      ) {
+        return path.join(
+          workspaceRootPath,
+          "node_modules",
+          "@storm-software",
+          "tsdoc",
+          "dist",
+          configFilePath
+        );
+      }
+
+      return undefined;
+    }
+
+    static getStormConfigFile(options) {
+        const type = options && options.type
+          ? options.type
+          : "recommended";
+        const configFile = options && options.configFile
+          ? options.configFile
+          : undefined;
+
+        let workspaceRoot = "";
+        if (process.env.STORM_WORKSPACE_ROOT || process.env.NX_WORKSPACE_ROOT_PATH) {
+          workspaceRoot = process.env.STORM_WORKSPACE_ROOT || process.env.NX_WORKSPACE_ROOT_PATH;
+        }
+        if (!workspaceRoot) {
+          workspaceRoot = findWorkspaceRoot();
+        }
+
+        let currentConfigFile = process.env.STORM_TSDOC_CONFIG_FILE
+          ? process.env.STORM_TSDOC_CONFIG_FILE
+          : configFile;
+        Debug_1.Debug.log(`Config file from user options: ${currentConfigFile || "(none)"}`);
+
+        let configFilePath = ConfigCache._checkConfigFilePath(
+          workspaceRoot,
+          currentConfigFile
+        );
+        if (!configFilePath) {
+          currentConfigFile = `${((
+              process.env.STORM_TSDOC_TYPE
+                ? process.env.STORM_TSDOC_TYPE
+                : type
+            ) || "recommended").replace(/\.jsonc?$/, "")}.json`;
+          Debug_1.Debug.log(`Config file from user options not found, trying: ${currentConfigFile}`);
+
+          configFilePath = ConfigCache._checkConfigFilePath(
+            workspaceRoot,
+            currentConfigFile
+          );
+          if (!configFilePath) {
+            configFilePath = ConfigCache._checkConfigFilePath(
+              workspaceRoot,
+              "recommended.json"
+            );
+            if (!configFilePath) {
+              configFilePath = ConfigCache._checkConfigFilePath(
+                workspaceRoot,
+                "tsdoc.json"
+              );
+            }
+          }
+        }
+
+        const cacheKey = configFilePath || "missing-config-file";
+        Debug_1.Debug.log(`Storm Software Configuration Cache key: "${cacheKey}"`);
+
+        const nowMs = ConfigCache._getTimeInMs();
+        let cachedConfig = undefined;
+
+        // Do we have a cached object?
+        cachedConfig = ConfigCache._cachedConfigs.get(cacheKey);
+        if (cachedConfig) {
+            Debug_1.Debug.log('Cache hit');
+
+            // Is the cached object still valid?
+            const loadAgeMs = nowMs - cachedConfig.loadTimeMs;
+            const lastCheckAgeMs = nowMs - cachedConfig.lastCheckTimeMs;
+            if (loadAgeMs > CACHE_EXPIRE_MS || loadAgeMs < 0) {
+                Debug_1.Debug.log('Evicting because item is expired');
+
+                cachedConfig = undefined;
+                ConfigCache._cachedConfigs.delete(cacheKey);
+            } else if (lastCheckAgeMs > CACHE_CHECK_INTERVAL_MS || lastCheckAgeMs < 0) {
+                Debug_1.Debug.log('Checking for modifications');
+
+                cachedConfig.lastCheckTimeMs = nowMs;
+                if (cachedConfig.configFile.checkForModifiedFiles()) {
+                    // Invalidate the cache because it failed to load completely
+                    Debug_1.Debug.log('Evicting because item was modified');
+                    cachedConfig = undefined;
+                    ConfigCache._cachedConfigs.delete(cacheKey);
+                }
+            }
+        }
+
         // Load the object
         if (!cachedConfig) {
             if (ConfigCache._cachedConfigs.size > CACHE_MAX_SIZE) {
                 Debug_1.Debug.log('Clearing cache');
                 ConfigCache._cachedConfigs.clear(); // avoid a memory leak
             }
+
             const configFile = tsdoc_config_1.TSDocConfigFile.loadFile(configFilePath);
             if (configFile.fileNotFound) {
                 Debug_1.Debug.log(`File not found: "${configFilePath}"`);
@@ -94,6 +373,7 @@ class ConfigCache {
             else {
                 Debug_1.Debug.log(`Loaded: "${configFilePath}"`);
             }
+
             cachedConfig = {
                 configFile,
                 lastCheckTimeMs: nowMs,
@@ -101,6 +381,7 @@ class ConfigCache {
             };
             ConfigCache._cachedConfigs.set(cacheKey, cachedConfig);
         }
+
         return cachedConfig.configFile;
     }
 }
diff --git a/lib/Debug.js b/lib/Debug.js
index abdc844796e31fa49872c8eed312b0a19299e1ef..7463bad88594dd18a25e1e4a9ae36318eb6bc4d4 100644
--- a/lib/Debug.js
+++ b/lib/Debug.js
@@ -6,8 +6,9 @@ exports.Debug = void 0;
 class Debug {
     // To debug the plugin, temporarily uncomment the body of this function
     static log(message) {
-        // message = require("process").pid + ": " + message;
-        // console.log(message);
+        message = require("process").pid + ": " + message;
+        console.log(message);
+
         // require('fs').writeFileSync('C:\\Git\\log.txt', message + '\r\n', { flag: 'as' });
     }
 }
diff --git a/lib/index.js b/lib/index.js
index 73c357f96e3c69ac87e19cc3e5d2733073d6aef3..36e90b80f02414eccbe9f562fe91f5e2136ba809 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -15,48 +15,132 @@ const plugin = {
         // from the NPM package name, and then appending this string.
         syntax: {
             meta: {
-                messages: Object.assign({ 'error-loading-config-file': 'Error loading TSDoc config file:\n{{details}}', 'error-applying-config': 'Error applying TSDoc configuration: {{details}}' }, tsdocMessageIds),
+                messages: Object.assign({
+                  'error-loading-storm-config': 'Error loading Storm Software\'s shared TSDoc config file:\n{{details}} \n\nPlease ensure "@storm-software/tsdoc" is installed in the workspace root.',
+                  'error-loading-config-file': 'Error loading TSDoc config file:\n{{details}}',
+                  'error-applying-storm-config': 'Error applying Storm Software\'s shared TSDoc configuration: {{details}}',
+                  'error-applying-config': 'Error applying TSDoc configuration: {{details}}'
+                }, tsdocMessageIds),
                 type: 'problem',
                 docs: {
                     description: 'Validates that TypeScript documentation comments conform to the TSDoc standard',
                     category: 'Stylistic Issues',
-                    // This package is experimental
-                    recommended: false,
+                    recommended: true,
                     url: 'https://tsdoc.org/pages/packages/eslint-plugin-tsdoc'
-                }
+                },
+                schema: [
+                  {
+                    type: "object",
+                    properties: {
+                      type: {
+                        type: "string",
+                        enum: ["typedoc", "api-extractor", "base", "core", "callouts", "recommended"],
+                        description:
+                          "The TSDoc configuration type to use from '@storm-software/tsdoc'. Defaults to 'recommended' (@storm-software/tsdoc/recommended.json)."
+                      },
+                      configFile: {
+                        type: "string",
+                        description:
+                          "The TSDoc configuration file to use."
+                      }
+                    },
+                    additionalProperties: false
+                  }
+                ],
             },
+            defaultOptions: [
+              {
+                type: "recommended"
+              }
+            ],
             create: (context) => {
-                const sourceFilePath = context.filename;
+                const tsdocConfiguration = new tsdoc_1.TSDocConfiguration();
+
                 // If eslint is configured with @typescript-eslint/parser, there is a parser option
                 // to explicitly specify where the tsconfig file is. Use that if available.
                 const tsConfigDir = context.parserOptions.tsconfigRootDir;
+                const sourceFilePath = context.filename;
+
                 Debug_1.Debug.log(`Linting: "${sourceFilePath}"`);
-                const tsdocConfiguration = new tsdoc_1.TSDocConfiguration();
+
                 try {
-                    const tsdocConfigFile = ConfigCache_1.ConfigCache.getForSourceFile(sourceFilePath, tsConfigDir);
+                  let tsdocConfigFile = {};
+                  try {
+                    const userOptions = context.options
+                      ? Array.isArray(context.options)
+                        ? context.options.length > 0
+                          ? context.options[0]
+                          : {}
+                        : context.options
+                      : {};
+
+                    const config = { type: userOptions.type || "recommended", configFile: userOptions.configFile };
+                    Debug_1.Debug.log(`Using eslint-plugin-tsdoc config: ${JSON.stringify(config, undefined, 2)}`);
+
+                    tsdocConfigFile = ConfigCache_1.ConfigCache.getStormConfigFile(config);
                     if (!tsdocConfigFile.fileNotFound) {
-                        if (tsdocConfigFile.hasErrors) {
-                            context.report({
-                                loc: { line: 1, column: 1 },
-                                messageId: 'error-loading-config-file',
-                                data: {
-                                    details: tsdocConfigFile.getErrorSummary()
-                                }
-                            });
-                        }
-                        try {
-                            tsdocConfigFile.configureParser(tsdocConfiguration);
-                        }
-                        catch (e) {
-                            context.report({
-                                loc: { line: 1, column: 1 },
-                                messageId: 'error-applying-config',
-                                data: {
-                                    details: e.message
-                                }
-                            });
-                        }
+                      if (tsdocConfigFile.hasErrors) {
+                          context.report({
+                              loc: { line: 1, column: 1 },
+                              messageId: 'error-loading-storm-config',
+                              data: {
+                                  details: tsdocConfigFile.getErrorSummary()
+                              }
+                          });
+                      }
+
+                      try {
+                          tsdocConfigFile.configureParser(tsdocConfiguration);
+                      }
+                      catch (e) {
+                          context.report({
+                              loc: { line: 1, column: 1 },
+                              messageId: 'error-applying-storm-config',
+                              data: {
+                                  details: e.message
+                              }
+                          });
+                      }
+                    }
+                  } catch (e) {
+                    context.report({
+                      loc: { line: 1, column: 1 },
+                      messageId: 'error-loading-storm-config',
+                      data: {
+                          details: `Unexpected exception: ${e.message}`
+                      }
+                    });
+                  }
+
+                  if (!tsdocConfigFile || tsdocConfigFile.fileNotFound || tsdocConfigFile.hasErrors) {
+                    // If the Storm Software config was not found, or if it has errors, try to load the config file
+                    tsdocConfigFile = ConfigCache_1.ConfigCache.getForSourceFile(sourceFilePath, tsConfigDir);
+
+                    if (!tsdocConfigFile.fileNotFound) {
+                      if (tsdocConfigFile.hasErrors) {
+                        context.report({
+                            loc: { line: 1, column: 1 },
+                            messageId: 'error-loading-config-file',
+                            data: {
+                                details: tsdocConfigFile.getErrorSummary()
+                            }
+                        });
+                      }
+
+                      try {
+                        tsdocConfigFile.configureParser(tsdocConfiguration);
+                      }
+                      catch (e) {
+                        context.report({
+                            loc: { line: 1, column: 1 },
+                            messageId: 'error-applying-config',
+                            data: {
+                                details: e.message
+                            }
+                        });
+                      }
                     }
+                  }
                 }
                 catch (e) {
                     context.report({
diff --git a/package.json b/package.json
index f616530237bf40b93d931e8fd2f19704e829785b..7ba5a40183030bb79db76229a9b8a372fabf6f7f 100644
--- a/package.json
+++ b/package.json
@@ -21,9 +21,18 @@
   "homepage": "https://tsdoc.org/",
   "main": "lib/index.js",
   "typings": "lib/index.d.ts",
+  "peerDependencies": {
+    "@storm-software/tsdoc": "*"
+  },
+  "peerDependenciesMeta": {
+    "@storm-software/tsdoc": {
+      "optional": false
+    }
+  },
   "dependencies": {
     "@microsoft/tsdoc": "0.15.1",
-    "@microsoft/tsdoc-config": "0.17.1"
+    "@microsoft/tsdoc-config": "0.17.1",
+    "@storm-software/config-tools": "latest"
   },
   "devDependencies": {
     "@rushstack/heft-node-rig": "~2.6.11",
