diff --git a/src/appendStep.js b/src/appendStep.js
index b73304dbdc36bd0588ff06a0eef7ce3e35b76c6e..e10ec85cb681a97a5f317078745a762019ac3147 100644
--- a/src/appendStep.js
+++ b/src/appendStep.js
@@ -56,7 +56,7 @@ const appendStep = (
           );
         }
 
-        const plugin = require(pluginName);
+        const plugin = await import(pluginName);
         const step = plugin && plugin[stepName];
 
         if (!step) {
diff --git a/src/wrapStep.js b/src/wrapStep.js
index 2594560bd174c32060a305c47ab2e9fd7ded2b18..41ad3e0f2020a8823daca8059cb01fd30df9ff2b 100644
--- a/src/wrapStep.js
+++ b/src/wrapStep.js
@@ -46,7 +46,7 @@ const wrapStep = (
           );
         }
 
-        const plugin = require(pluginName);
+        const plugin = await import(pluginName);
         const step = plugin && plugin[stepName];
 
         if (!step) {
