diff --git a/lib/get-config.js b/lib/get-config.js
index 429577e1a4d840f3d73d53e76a2f7fc25e3d6918..f72928e5479846355c7c0b057e911089e804f67b 100644
--- a/lib/get-config.js
+++ b/lib/get-config.js
@@ -14,7 +14,6 @@ import { parseConfig, validatePlugin } from "./plugins/utils.js";
 
 const debug = debugConfig("semantic-release:config");
 const __dirname = dirname(fileURLToPath(import.meta.url));
-const require = createRequire(import.meta.url);
 
 const CONFIG_NAME = "release";
 
@@ -35,7 +34,7 @@ export default async (context, cliOptions) => {
     options = {
       ...(await castArray(extendPaths).reduce(async (eventualResult, extendPath) => {
         const result = await eventualResult;
-        const extendsOptions = require(resolveFrom.silent(__dirname, extendPath) || resolveFrom(cwd, extendPath));
+        const extendsOptions = await import(resolveFrom.silent(__dirname, extendPath) || resolveFrom(cwd, extendPath));
 
         // For each plugin defined in a shareable config, save in `pluginsPath` the extendable config path,
         // so those plugin will be loaded relative to the config file
