diff --git a/dist/index.js b/dist/index.js
index 68c197cb2b8dfa928a448475b4020e7301da959e..426559111d4475c7e4fecc69217d037e9ff0c516 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1605,8 +1605,30 @@ async function runEsbuild(options, {
   buildDependencies,
   pluginContainer
 }) {
-  const pkg = await _chunk52KWB5QBjs.loadPkg.call(void 0, process.cwd());
-  const deps = await _chunk52KWB5QBjs.getProductionDeps.call(void 0, process.cwd());
+  let pkgPath = options.outDir 
+    ? options.outDir
+    : process.cwd();
+  console.error("Original pkgPath: ", pkgPath);
+  if (pkgPath) {
+    const splits = pkgPath.split(_path2.default.sep);
+    console.error("splits: ", splits);
+    if (splits.length > 0) {
+     const pkgPathIndex = splits.slice(0, splits.length)
+      .findLastIndex(index => 
+        existsSync(_path2.default.join(splits.slice(0, index)
+          .join(_path2.default.sep), "package.json")));
+          console.error("pkgPathIndex: ", pkgPathIndex);
+      if (pkgPathIndex) {
+        pkgPath = splits.slice(0, pkgPathIndex).join(_path2.default.sep);
+        console.error("Updated pkgPath: ", pkgPath);
+      }
+    }
+  }
+
+  console.error("Final pkgPath: ", pkgPath);
+  const pkg = await _chunk52KWB5QBjs.loadPkg.call(void 0, pkgPath);
+  console.error("pkg: ", pkg);
+  const deps = await _chunk52KWB5QBjs.getProductionDeps.call(void 0, pkgPath);
   const external = [
     // Exclude dependencies, e.g. `lodash`, `lodash/get`
     ...deps.map((dep) => new RegExp(`^${dep}($|\\/|\\\\)`)),
@@ -2052,6 +2074,7 @@ var sizeReporter = () => {
 // src/plugins/tree-shaking.ts
 var import_rollup_plugin_hashbang = _chunkB7UJSB4Njs.__toESM.call(void 0, _chunk52KWB5QBjs.require_dist.call(void 0, ));
 var _rollup = require('rollup');
+const { exists } = require('fs-extra');
 var treeShakingPlugin = ({
   treeshake,
   name,
