diff --git a/dist/index.js b/dist/index.js
index 68c197cb2b8dfa928a448475b4020e7301da959e..05f0e52b1e2d514b704f4f4e48507237c90cd386 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1605,8 +1605,31 @@ async function runEsbuild(options, {
   buildDependencies,
   pluginContainer
 }) {
-  const pkg = await _chunk52KWB5QBjs.loadPkg.call(void 0, process.cwd());
-  const deps = await _chunk52KWB5QBjs.getProductionDeps.call(void 0, process.cwd());
+  let pkgPath = options.outDir 
+    ? options.outDir
+    : process.cwd();
+
+  logger.info(format, `⚡ Preparing to run ESBuild: ${pkgPath}`);
+  if (pkgPath) {
+    const splits = pkgPath.split(_path2.default.sep);
+    if (splits.length > 0) {
+      const pkgPathIndex = splits.slice(0, 
+          splits.length).reverse()
+        .findIndex((split, index) => 
+          existsSync(_path2.default.join(splits.slice(0, 
+            index), 
+              "package.json")));
+
+      logger.info(format, `⚡ Package path index: ${pkgPathIndex}`);
+      if (pkgPathIndex) {
+        pkgPath = splits.slice(0, pkgPathIndex).join(_path2.default.sep);
+      }
+    }
+  }
+
+  logger.info(format, `⚡ Running ESBuild: ${pkgPath}`);
+  const pkg = await _chunk52KWB5QBjs.loadPkg.call(void 0, pkgPath);
+  const deps = await _chunk52KWB5QBjs.getProductionDeps.call(void 0, pkgPath);
   const external = [
     // Exclude dependencies, e.g. `lodash`, `lodash/get`
     ...deps.map((dep) => new RegExp(`^${dep}($|\\/|\\\\)`)),
@@ -1676,7 +1699,7 @@ async function runEsbuild(options, {
       jsxFragment: options.jsxFragment,
       sourcemap: options.sourcemap ? "external" : false,
       target: options.target,
-      banner,
+      banner: { js: "Created by Storm Software", css: "Created by Storm Software" },
       footer,
       tsconfig: options.tsconfig,
       loader: {
@@ -1727,7 +1750,7 @@ async function runEsbuild(options, {
       outExtension: options.legacyOutput ? void 0 : outExtension,
       write: false,
       splitting,
-      logLevel: "error",
+      logLevel: 'info',
       minify: options.minify === "terser" ? false : options.minify,
       minifyWhitespace: options.minifyWhitespace,
       minifyIdentifiers: options.minifyIdentifiers,
@@ -2052,6 +2075,7 @@ var sizeReporter = () => {
 // src/plugins/tree-shaking.ts
 var import_rollup_plugin_hashbang = _chunkB7UJSB4Njs.__toESM.call(void 0, _chunk52KWB5QBjs.require_dist.call(void 0, ));
 var _rollup = require('rollup');
+const { exists } = require('fs-extra');
 var treeShakingPlugin = ({
   treeshake,
   name,
diff --git a/dist/rollup.js b/dist/rollup.js
index 0f6400eedfad49091ca952ee5863bd027e3b8417..f06f35c5fb7e104498aac9a18bb3c030093e70ae 100644
--- a/dist/rollup.js
+++ b/dist/rollup.js
@@ -6805,6 +6805,13 @@ export { ${[...exportedNames].join(", ")} };
             }
           }
         }
+
+        // https://github.com/Swatinem/rollup-plugin-dts/pull/287
+        // `this` is a reserved keyword that retrains meaning in certain Type-only contexts, including classes
+        if (name === "this") {
+          return;
+        }
+
         const { ident, expr } = createReference(id);
         this.declaration.params.push(expr);
         this.returnExpr.elements.push(ident);
